### ISSUE TITLE
About empty element

### ISSUE THREAD

**Issue by swtv-kaist** [swtv-kaist/cs458-fall22#2]
(taken from naryeong kim's post on KLMS)
I think the nodes after last node should be empty to satisfy the shape property.
But if the nodes can have non-positive integer, what is the value of empty node instead of zero? 

Or can we think that last node is the last element of array?

**Comment by swtv-kaist** [comment]
> what is the value of empty node instead of zero? 

A value of an empty node does NOT matter (i.e., it can be any non-deterministic value) for verifying `max_heapify`.
This is because `max_heapify ` checks if a given node is empty or not by checking its index and `h_size`.

**Comment by KimNaRyeong** [comment]
> checking its index and `h_size`

I didn't understand yet.
For example, let the index that violates the property be 2. Then is maxheap = [0, 3, 5, 6, 7, empty, empty, empty] possible?
Or should all elements of after the index be filled with integer?

**Comment by swtv-kaist** [comment]
> let the index that violates the property be 2. Then is maxheap = [0, 3, 5, 6, 7, empty, empty, empty] possible?

Yes. Your maxheap example w/ `h_size`=4 is a valid input to `max_heapify(...2...)`

**Comment by swtv-kaist** [comment]
test2

Sincerely
-Moonzoo Kim
----------------------------------------
Moonzoo Kim, Ph.D
RM# 2434 E3-1 School of Computing KAIST
373-1 Guseong-dong, Yuseong-gu
Daejeon, South Korea 34141
Phone:+82-42-350-3543
Fax:+82-42-350-3510
http://swtv.kaist.ac.kr/members/mzkim
----------------------------------------
